common/contexts/src/dal/dal_context.rs

```diff
@@ -66,13 +73,13 @@ impl Layer for DalContext {
 #[async_trait]
 impl Accessor for DalContext {
     async fn create(&self, args: &OpCreate) -> Result<()> {
-        self.inner.as_ref().unwrap().create(args).await
+        self.get_inner().create(args).await
```

Comment:
Since `async fn create(&self, args: &OpCreate)` returns `Result<()>`,  will it be better to make `get_inner`  return `Result<Arc<dyn Accessor>>`? such that the `panic!` can be avoided.

Panics can be caught and handled by the session gracefully though (hope I get it right)... 

@zhang2014  is `painc`  suggested being used here? (`get_inner` may panic)

Reply:
I use `panic` here because it never should happen unless we use the `DalContext` wrongly which should be fixed ASAP.

Reply:
> Panics can be caught and handled by the session gracefully though (hope I get it right)...

No, I think it never happens.

The only way to initiate the `DalContext` is `DalContext::new()` or `DalContext::layer()` . Maybe I should also remove the `Default` too?

Reply:
`QueryContextShared::try_create()` will init a `dal_context` with `Default::default()`.

@dantengsky I got your concerns. Let's return an error instead.

Reply:
> Maybe I should also remove the `Default` too?

:+1:  I tried a little bit, but failed to figure out how to implement the `QueryContextShared::try_create`

Reply:
Yep, I met the same question so I realized that it's better to return errors instead.